Intermittent test failure - Class org.mozilla.fenix.shortcut.PwaOnboardingObserverTest

```
java.lang.AssertionError: Verification failed: call 5 of 5: class org.mozilla.fenix.components.AnalyticsKt.isSentryEnabled()) should not be called

Calls:
1) class org.mozilla.fenix.components.AnalyticsKt.isSentryEnabled()
2) class org.mozilla.fenix.components.AnalyticsKt.isSentryEnabled()


Stack traces:
1)                                                                 io.mockk.impl.InternalPlatform.captureStackTrace                                                                                 (InternalPlatform.kt:121)                  
                                                                     io.mockk.impl.stub.MockKStub.handleInvocation                                                                                  (MockKStub.kt:247)                         
                                   io.mockk.impl.instantiation.JvmMockFactoryHelper$mockHandler$1.invocation                                                                                        (JvmMockFactoryHelper.kt:25)               
                                                            io.mockk.proxy.jvm.advice.Interceptor.call                                                                                              (Interceptor.kt:20)                        
                                                         org.mozilla.fenix.components.AnalyticsKt.isSentryEnabled                                                                                   (Analytics.kt:144)                         
                                                            org.mozilla.fenix.ext.NavControllerKt.recordIdException                                                                                 (NavController.kt:31)                      
                                                            org.mozilla.fenix.ext.NavControllerKt.nav                                                                                               (NavController.kt:22)                      
                                                            org.mozilla.fenix.ext.NavControllerKt.nav$default                                                                                       (NavController.kt:18)                      
        org.mozilla.fenix.shortcut.PwaOnboardingObserver$start$1$invokeSuspend$$inlined$collect$1.emit                                                                                              (Collect.kt:138)                           
      mozilla.components.support.ktx.kotlinx.coroutines.flow.FlowKt$ifChanged$$inlined$filter$1$2.emit                                                                                              (Collect.kt:136)                           
   org.mozilla.fenix.shortcut.PwaOnboardingObserver$start$1$invokeSuspend$$inlined$mapNotNull$1$2.emit                                                                                              (Collect.kt:136)                           
                                                       kotlinx.coroutines.flow.FlowKt__ChannelsKt.emitAllImpl$FlowKt__ChannelsKt                                                                    (Channels.kt:61)                           
                                                       kotlinx.coroutines.flow.FlowKt__ChannelsKt.emitAll                                                                                           (Channels.kt:30)                           
                                                                   kotlinx.coroutines.flow.FlowKt.emitAll                                                                                           (-:1)                                      
                                           kotlinx.coroutines.flow.internal.ChannelFlow$collect$2.invokeSuspend                                                                                     (ChannelFlow.kt:135)                       
                                           kotlinx.coroutines.flow.internal.ChannelFlow$collect$2.invoke                                                                                            (ChannelFlow.kt:-1)                        
                                                     kotlinx.coroutines.intrinsics.UndispatchedKt.startUndispatchedOrReturn                                                                         (Undispatched.kt:91)                       
                                                              kotlinx.coroutines.CoroutineScopeKt.coroutineScope                                                                                    (CoroutineScope.kt:194)                    
                                                     kotlinx.coroutines.flow.internal.ChannelFlow.collect$suspendImpl                                                                               (ChannelFlow.kt:134)                       
                                                     kotlinx.coroutines.flow.internal.ChannelFlow.collect                                                                                           (ChannelFlow.kt:-1)                        
     org.mozilla.fenix.shortcut.PwaOnboardingObserver$start$1$invokeSuspend$$inlined$mapNotNull$1.collect                                                                                           (SafeCollector.common.kt:114)              
        mozilla.components.support.ktx.kotlinx.coroutines.flow.FlowKt$ifChanged$$inlined$filter$1.collect                                                                                           (SafeCollector.common.kt:114)              
                                         org.mozilla.fenix.shortcut.PwaOnboardingObserver$start$1.invokeSuspend                                                                                     (PwaOnboardingObserver.kt:71)              
                                         org.mozilla.fenix.shortcut.PwaOnboardingObserver$start$1.invoke                                                                                            (PwaOnboardingObserver.kt:-1)              
            mozilla.components.lib.state.ext.StoreExtensionsKt$flowScoped$$inlined$apply$lambda$1.invokeSuspend                                                                                     (StoreExtensions.kt:204)                   
                                              kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith                                                                                        (ContinuationImpl.kt:33)                   
                                                                kotlinx.coroutines.DispatchedTask.run                                                                                               (DispatchedTask.kt:106)                    
                                                  kotlinx.coroutines.test.TestCoroutineDispatcher.dispatch                                                                                          (TestCoroutineDispatcher.kt:50)            
                                              kotlinx.coroutines.test.internal.TestMainDispatcher.dispatch                                                                                          (MainTestDispatcher.kt:35)                 
                                             kotlinx.coroutines.internal.DispatchedContinuationKt.resumeCancellableWith                                                                             (DispatchedContinuation.kt:305)            
                                                      kotlinx.coroutines.intrinsics.CancellableKt.startCoroutineCancellable                                                                         (Cancellable.kt:30)                        
                                                      kotlinx.coroutines.intrinsics.CancellableKt.startCoroutineCancellable$default                                                                 (Cancellable.kt:27)                        
                                                                kotlinx.coroutines.CoroutineStart.invoke                                                                                            (CoroutineStart.kt:109)                    
                                                             kotlinx.coroutines.AbstractCoroutine.start                                                                                             (AbstractCoroutine.kt:158)                 
                                                 kotlinx.coroutines.BuildersKt__Builders_commonKt.launch                                                                                            (Builders.common.kt:56)                    
                                                                    kotlinx.coroutines.BuildersKt.launch                                                                                            (-:1)                                      
                                                 kotlinx.coroutines.BuildersKt__Builders_commonKt.launch$default                                                                                    (Builders.common.kt:49)                    
                                                                    kotlinx.coroutines.BuildersKt.launch$default                                                                                    (-:1)                                      
                                               mozilla.components.lib.state.ext.StoreExtensionsKt.flowScoped                                                                                        (StoreExtensions.kt:203)                   
                                                 org.mozilla.fenix.shortcut.PwaOnboardingObserver.start                                                                                             (PwaOnboardingObserver.kt:39)              
                                             org.mozilla.fenix.shortcut.PwaOnboardingObserverTest.GIVEN cfr should be shown WHEN installable page is loaded THEN we navigate to onboarding fragment (PwaOnboardingObserverTest.kt:95)          
                                                             sun.reflect.NativeMethodAccessorImpl.invoke0                                                                                           (NativeMethodAccessorImpl.java:-2)N        
                                                             sun.reflect.NativeMethodAccessorImpl.invoke                                                                                            (NativeMethodAccessorImpl.java:62)         
                                                         sun.reflect.DelegatingMethodAccessorImpl.invoke                                                                                            (DelegatingMethodAccessorImpl.java:43)     
                                                                         java.lang.reflect.Method.invoke                                                                                            (Method.java:498)                          
                                                        org.junit.runners.model.FrameworkMethod$1.runReflectiveCall                                                                                 (FrameworkMethod.java:50)                  
                                              org.junit.internal.runners.model.ReflectiveCallable.run                                                                                               (ReflectiveCallable.java:12)               
                                                          org.junit.runners.model.FrameworkMethod.invokeExplosively                                                                                 (FrameworkMethod.java:47)                  
                                               org.junit.internal.runners.statements.InvokeMethod.evaluate                                                                                          (InvokeMethod.java:17)                     
                                                 org.junit.internal.runners.statements.RunBefores.evaluate                                                                                          (RunBefores.java:26)                       
                                                  org.junit.internal.runners.statements.RunAfters.evaluate                                                                                          (RunAfters.java:27)                        
                                                                    org.junit.rules.TestWatcher$1.evaluate                                                                                          (TestWatcher.java:55)                      
                                                                         org.junit.rules.RunRules.evaluate                                                                                          (RunRules.java:20)                         
                                                                   org.junit.runners.ParentRunner.runLeaf                                                                                           (ParentRunner.java:325)                    
                                                         org.junit.runners.BlockJUnit4ClassRunner.runChild                                                                                          (BlockJUnit4ClassRunner.java:78)           
                                                         org.junit.runners.BlockJUnit4ClassRunner.runChild                                                                                          (BlockJUnit4ClassRunner.java:57)           
                                                                 org.junit.runners.ParentRunner$3.run                                                                                               (ParentRunner.java:290)                    
                                                                 org.junit.runners.ParentRunner$1.schedule                                                                                          (ParentRunner.java:71)                     
                                                                   org.junit.runners.ParentRunner.runChildren                                                                                       (ParentRunner.java:288)                    
                                                                   org.junit.runners.ParentRunner.access$000                                                                                        (ParentRunner.java:58)                     
                                                                 org.junit.runners.ParentRunner$2.evaluate                                                                                          (ParentRunner.java:268)                    
                                                                   org.junit.runners.ParentRunner.run                                                                                               (ParentRunner.java:363)                    
                               org.gradle.api.internal.tasks.testing.junit.JUnitTestClassExecutor.runTestClass                                                                                      (JUnitTestClassExecutor.java:110)          
                               org.gradle.api.internal.tasks.testing.junit.JUnitTestClassExecutor.execute                                                                                           (JUnitTestClassExecutor.java:58)           
                               org.gradle.api.internal.tasks.testing.junit.JUnitTestClassExecutor.execute                                                                                           (JUnitTestClassExecutor.java:38)           
                      org.gradle.api.internal.tasks.testing.junit.AbstractJUnitTestClassProcessor.processTestClass                                                                                  (AbstractJUnitTestClassProcessor.java:62)  
                                    org.gradle.api.internal.tasks.testing.SuiteTestClassProcessor.processTestClass                                                                                  (SuiteTestClassProcessor.java:51)          
                                                            sun.reflect.GeneratedMethodAccessor88.invoke                                                                                            (-:-1)                                     
                                                         sun.reflect.DelegatingMethodAccessorImpl.invoke                                                                                            (DelegatingMethodAccessorImpl.java:43)     
                                                                         java.lang.reflect.Method.invoke                                                                                            (Method.java:498)                          
                                                  org.gradle.internal.dispatch.ReflectionDispatch.dispatch                                                                                          (ReflectionDispatch.java:36)               
                                                  org.gradle.internal.dispatch.ReflectionDispatch.dispatch                                                                                          (ReflectionDispatch.java:24)               
                                          org.gradle.internal.dispatch.ContextClassLoaderDispatch.dispatch                                                                                          (ContextClassLoaderDispatch.java:33)       
                   org.gradle.internal.dispatch.ProxyDispatchAdapter$DispatchingInvocationHandler.invoke                                                                                            (ProxyDispatchAdapter.java:94)             
                                                                            com.sun.proxy.$Proxy2.processTestClass                                                                                  (-:-1)                                     
                                          org.gradle.api.internal.tasks.testing.worker.TestWorker.processTestClass                                                                                  (TestWorker.java:119)                      
                                                            sun.reflect.GeneratedMethodAccessor87.invoke                                                                                            (-:-1)                                     
                                                         sun.reflect.DelegatingMethodAccessorImpl.invoke                                                                                            (DelegatingMethodAccessorImpl.java:43)     
                                                                         java.lang.reflect.Method.invoke                                                                                            (Method.java:498)                          
                                                  org.gradle.internal.dispatch.ReflectionDispatch.dispatch                                                                                          (ReflectionDispatch.java:36)               
                                                  org.gradle.internal.dispatch.ReflectionDispatch.dispatch                                                                                          (ReflectionDispatch.java:24)               
         org.gradle.internal.remote.internal.hub.MessageHubBackedObjectConnection$DispatchWrapper.dispatch                                                                                          (MessageHubBackedObjectConnection.java:182)
         org.gradle.internal.remote.internal.hub.MessageHubBackedObjectConnection$DispatchWrapper.dispatch                                                                                          (MessageHubBackedObjectConnection.java:164)
                                       org.gradle.internal.remote.internal.hub.MessageHub$Handler.run                                                                                               (MessageHub.java:414)                      
                             org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute                                                                                         (ExecutorPolicy.java:64)                   
                                             org.gradle.internal.concurrent.ManagedExecutorImpl$1.run                                                                                               (ManagedExecutorImpl.java:48)              
                                                          java.util.concurrent.ThreadPoolExecutor.runWorker                                                                                         (ThreadPoolExecutor.java:1149)             
                                                   java.util.concurrent.ThreadPoolExecutor$Worker.run                                                                                               (ThreadPoolExecutor.java:624)              
                           org.gradle.internal.concurrent.ThreadFactoryImpl$ManagedThreadRunnable.run                                                                                               (ThreadFactoryImpl.java:56)                
                                                                                 java.lang.Thread.run                                                                                               (Thread.java:748)                          

2)                                                                 io.mockk.impl.InternalPlatform.captureStackTrace              (InternalPlatform.kt:121)           
                                                                     io.mockk.impl.stub.MockKStub.handleInvocation               (MockKStub.kt:247)                  
                                   io.mockk.impl.instantiation.JvmMockFactoryHelper$mockHandler$1.invocation                     (JvmMockFactoryHelper.kt:25)        
                                                            io.mockk.proxy.jvm.advice.Interceptor.call                           (Interceptor.kt:20)                 
                                                         org.mozilla.fenix.components.AnalyticsKt.isSentryEnabled                (Analytics.kt:144)                  
                                                            org.mozilla.fenix.ext.NavControllerKt.recordIdException              (NavController.kt:31)               
                                                            org.mozilla.fenix.ext.NavControllerKt.nav                            (NavController.kt:22)               
                                                            org.mozilla.fenix.ext.NavControllerKt.nav$default                    (NavController.kt:18)               
        org.mozilla.fenix.shortcut.PwaOnboardingObserver$start$1$invokeSuspend$$inlined$collect$1.emit                           (Collect.kt:138)                    
      mozilla.components.support.ktx.kotlinx.coroutines.flow.FlowKt$ifChanged$$inlined$filter$1$2.emit                           (Collect.kt:136)                    
   org.mozilla.fenix.shortcut.PwaOnboardingObserver$start$1$invokeSuspend$$inlined$mapNotNull$1$2.emit                           (Collect.kt:136)                    
                                                       kotlinx.coroutines.flow.FlowKt__ChannelsKt.emitAllImpl$FlowKt__ChannelsKt (Channels.kt:61)                    
                                         kotlinx.coroutines.flow.FlowKt__ChannelsKt$emitAllImpl$1.invokeSuspend                  (Channels.kt:-1)                    
                                              kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith                     (ContinuationImpl.kt:33)            
                                                                kotlinx.coroutines.DispatchedTask.run                            (DispatchedTask.kt:106)             
                                                  kotlinx.coroutines.test.TestCoroutineDispatcher.dispatch                       (TestCoroutineDispatcher.kt:50)     
                                              kotlinx.coroutines.test.internal.TestMainDispatcher.dispatch                       (MainTestDispatcher.kt:35)          
                                                              kotlinx.coroutines.DispatchedTaskKt.dispatch                       (DispatchedTask.kt:159)             
                                                   kotlinx.coroutines.CancellableContinuationImpl.dispatchResume                 (CancellableContinuationImpl.kt:362)
                                                   kotlinx.coroutines.CancellableContinuationImpl.completeResume                 (CancellableContinuationImpl.kt:479)
                                       kotlinx.coroutines.channels.AbstractChannel$ReceiveElement.completeResumeReceive          (AbstractChannel.kt:900)            
                                                         kotlinx.coroutines.channels.ArrayChannel.offerInternal                  (ArrayChannel.kt:84)                
                                                  kotlinx.coroutines.channels.AbstractSendChannel.send                           (AbstractChannel.kt:135)            
                                                     kotlinx.coroutines.channels.ChannelCoroutine.send$suspendImpl               (ChannelCoroutine.kt:-1)            
                                                     kotlinx.coroutines.channels.ChannelCoroutine.send                           (ChannelCoroutine.kt:-1)            
                       mozilla.components.lib.state.ext.StoreExtensionsKt$flow$1$subscription$1$1.invokeSuspend                  (StoreExtensions.kt:169)            
                                              kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith                     (ContinuationImpl.kt:33)            
                                                                kotlinx.coroutines.DispatchedTask.run                            (DispatchedTask.kt:106)             
                                                             kotlinx.coroutines.EventLoopImplBase.processNextEvent               (EventLoop.common.kt:274)           
                                                             kotlinx.coroutines.BlockingCoroutine.joinBlocking                   (Builders.kt:84)                    
                                                        kotlinx.coroutines.BuildersKt__BuildersKt.runBlocking                    (Builders.kt:59)                    
                                                                    kotlinx.coroutines.BuildersKt.runBlocking                    (-:1)                               
                                                        kotlinx.coroutines.BuildersKt__BuildersKt.runBlocking$default            (Builders.kt:38)                    
                                                                    kotlinx.coroutines.BuildersKt.runBlocking$default            (-:1)                               
                         mozilla.components.lib.state.ext.StoreExtensionsKt$flow$1$subscription$1.invoke                         (StoreExtensions.kt:169)            
                         mozilla.components.lib.state.ext.StoreExtensionsKt$flow$1$subscription$1.invoke                         (StoreExtensions.kt:-1)             
                                                  mozilla.components.lib.state.Store$Subscription.dispatch$lib_state_release     (Store.kt:152)                      
                                                               mozilla.components.lib.state.Store.transitionTo$lib_state_release (Store.kt:105)                      
                          mozilla.components.lib.state.internal.ReducerChainBuilder$build$chain$1.invoke                         (ReducerChainBuilder.kt:52)         
                          mozilla.components.lib.state.internal.ReducerChainBuilder$build$chain$1.invoke                         (ReducerChainBuilder.kt:18)         
                    mozilla.components.lib.state.internal.ReducerChainBuilder$build$threadCheck$1.invoke                         (ReducerChainBuilder.kt:57)         
                    mozilla.components.lib.state.internal.ReducerChainBuilder$build$threadCheck$1.invoke                         (ReducerChainBuilder.kt:18)         
        mozilla.components.lib.state.internal.ReducerChainBuilder$build$$inlined$forEach$lambda$1.invoke                         (ReducerChainBuilder.kt:62)         
        mozilla.components.lib.state.internal.ReducerChainBuilder$build$$inlined$forEach$lambda$1.invoke                         (ReducerChainBuilder.kt:18)         
                                                    mozilla.components.lib.state.Store$dispatch$1.invokeSuspend                  (Store.kt:91)                       
                                              kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith                     (ContinuationImpl.kt:33)            
                                                                kotlinx.coroutines.DispatchedTask.run                            (DispatchedTask.kt:106)             
                                                          java.util.concurrent.ThreadPoolExecutor.runWorker                      (ThreadPoolExecutor.java:1149)      
                                                   java.util.concurrent.ThreadPoolExecutor$Worker.run                            (ThreadPoolExecutor.java:624)       
                                                                                 java.lang.Thread.run                            (Thread.java:748)                   
	at io.mockk.impl.recording.states.VerifyingState.failIfNotPassed(VerifyingState.kt:66)
	at io.mockk.impl.recording.states.VerifyingState.recordingDone(VerifyingState.kt:42)
	at io.mockk.impl.recording.CommonCallRecorder.done(CommonCallRecorder.kt:47)
	at io.mockk.impl.eval.RecordedBlockEvaluator.record(RecordedBlockEvaluator.kt:60)
	at io.mockk.impl.eval.VerifyBlockEvaluator.verify(VerifyBlockEvaluator.kt:30)
	at io.mockk.MockKDsl.internalVerify(API.kt:118)
	at io.mockk.MockKKt.verify(MockK.kt:146)
	at io.mockk.MockKKt.verify$default(MockK.kt:143)
	at org.mozilla.fenix.shortcut.PwaOnboardingObserverTest.GIVEN web app is not installable WHEN page with manifest is loaded THEN nothing happens(PwaOnboardingObserverTest.kt:114)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
	at org.junit.rules.TestWatcher$1.evaluate(TestWatcher.java:55)
	at org.junit.rules.RunRules.evaluate(RunRules.java:20)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:57)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at org.gradle.api.internal.tasks.testing.junit.JUnitTestClassExecutor.runTestClass(JUnitTestClassExecutor.java:110)
	at org.gradle.api.internal.tasks.testing.junit.JUnitTestClassExecutor.execute(JUnitTestClassExecutor.java:58)
	at org.gradle.api.internal.tasks.testing.junit.JUnitTestClassExecutor.execute(JUnitTestClassExecutor.java:38)
	at org.gradle.api.internal.tasks.testing.junit.AbstractJUnitTestClassProcessor.processTestClass(AbstractJUnitTestClassProcessor.java:62)
	at org.gradle.api.internal.tasks.testing.SuiteTestClassProcessor.processTestClass(SuiteTestClassProcessor.java:51)
	at sun.reflect.GeneratedMethodAccessor88.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:36)
	at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:24)
	at org.gradle.internal.dispatch.ContextClassLoaderDispatch.dispatch(ContextClassLoaderDispatch.java:33)
	at org.gradle.internal.dispatch.ProxyDispatchAdapter$DispatchingInvocationHandler.invoke(ProxyDispatchAdapter.java:94)
	at com.sun.proxy.$Proxy2.processTestClass(Unknown Source)
	at org.gradle.api.internal.tasks.testing.worker.TestWorker.processTestClass(TestWorker.java:119)
	at sun.reflect.GeneratedMethodAccessor87.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:36)
	at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:24)
	at org.gradle.internal.remote.internal.hub.MessageHubBackedObjectConnection$DispatchWrapper.dispatch(MessageHubBackedObjectConnection.java:182)
	at org.gradle.internal.remote.internal.hub.MessageHubBackedObjectConnection$DispatchWrapper.dispatch(MessageHubBackedObjectConnection.java:164)
	at org.gradle.internal.remote.internal.hub.MessageHub$Handler.run(MessageHub.java:414)
	at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:64)
	at org.gradle.internal.concurrent.ManagedExecutorImpl$1.run(ManagedExecutorImpl.java:48)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at org.gradle.internal.concurrent.ThreadFactoryImpl$ManagedThreadRunnable.run(ThreadFactoryImpl.java:56)
	at java.lang.Thread.run(Thread.java:748)
```

--- a/app/src/main/java/org/mozilla/fenix/shortcut/PwaOnboardingObserver.kt
+++ b/app/src/main/java/org/mozilla/fenix/shortcut/PwaOnboardingObserver.kt
@@ -4,6 +4,7 @@
 
 package org.mozilla.fenix.shortcut
 
+import androidx.annotation.VisibleForTesting
 import androidx.lifecycle.LifecycleOwner
 import androidx.navigation.NavController
 import kotlinx.coroutines.CoroutineScope
@@ -47,9 +48,7 @@ class PwaOnboardingObserver(
                 if (webAppUseCases.isInstallable() && !settings.userKnowsAboutPwas) {
                     settings.incrementVisitedInstallableCount()
                     if (settings.shouldShowPwaCfr) {
-                        val directions =
-                            BrowserFragmentDirections.actionBrowserFragmentToPwaOnboardingDialogFragment()
-                        navController.nav(R.id.browserFragment, directions)
+                        navigateToPwaOnboarding()
                         settings.lastCfrShownTimeInMillis = System.currentTimeMillis()
                         settings.userKnowsAboutPwas = true
                     }
@@ -60,5 +59,13 @@ class PwaOnboardingObserver(
 
     fun stop() {
         scope?.cancel()
+    }
+
+    @VisibleForTesting
+    internal fun navigateToPwaOnboarding() {
+        navController.nav(
+            R.id.browserFragment,
+            BrowserFragmentDirections.actionBrowserFragmentToPwaOnboardingDialogFragment()
+        )
     }
 }

--- a/app/src/test/java/org/mozilla/fenix/shortcut/PwaOnboardingObserverTest.kt
+++ b/app/src/test/java/org/mozilla/fenix/shortcut/PwaOnboardingObserverTest.kt
@@ -10,6 +10,7 @@ import androidx.lifecycle.LifecycleRegistry
 import androidx.navigation.NavController
 import io.mockk.every
 import io.mockk.mockk
+import io.mockk.spyk
 import io.mockk.verify
 import kotlinx.coroutines.ExperimentalCoroutinesApi
 import kotlinx.coroutines.test.TestCoroutineDispatcher
@@ -24,9 +25,6 @@ import org.junit.After
 import org.junit.Before
 import org.junit.Rule
 import org.junit.Test
-import org.mozilla.fenix.R
-import org.mozilla.fenix.browser.BrowserFragmentDirections
-import org.mozilla.fenix.ext.nav
 import org.mozilla.fenix.utils.Settings
 
 @ExperimentalCoroutinesApi
@@ -54,17 +52,19 @@ class PwaOnboardingObserverTest {
             )
         )
         lifecycleOwner = MockedLifecycleOwner(Lifecycle.State.STARTED)
+
         navigationController = mockk(relaxed = true)
         settings = mockk(relaxed = true)
         webAppUseCases = mockk(relaxed = true)
 
-        pwaOnboardingObserver = PwaOnboardingObserver(
+        pwaOnboardingObserver = spyk(PwaOnboardingObserver(
             store = store,
             lifecycleOwner = lifecycleOwner,
             navController = navigationController,
             settings = settings,
             webAppUseCases = webAppUseCases
-        )
+        ))
+        every { pwaOnboardingObserver.navigateToPwaOnboarding() } returns Unit
     }
 
     @After
@@ -81,10 +81,7 @@ class PwaOnboardingObserverTest {
 
         store.dispatch(ContentAction.UpdateWebAppManifestAction("1", mockk())).joinBlocking()
         verify { settings.incrementVisitedInstallableCount() }
-        verify(exactly = 0) { navigationController.nav(
-            R.id.browserFragment,
-            BrowserFragmentDirections.actionBrowserFragmentToPwaOnboardingDialogFragment())
-        }
+        verify(exactly = 0) { pwaOnboardingObserver.navigateToPwaOnboarding() }
     }
 
     @Test
@@ -96,10 +93,7 @@ class PwaOnboardingObserverTest {
 
         store.dispatch(ContentAction.UpdateWebAppManifestAction("1", mockk())).joinBlocking()
         verify { settings.incrementVisitedInstallableCount() }
-        verify { navigationController.nav(
-            R.id.browserFragment,
-            BrowserFragmentDirections.actionBrowserFragmentToPwaOnboardingDialogFragment())
-        }
+        verify { pwaOnboardingObserver.navigateToPwaOnboarding() }
     }
 
     @Test
@@ -111,10 +105,7 @@ class PwaOnboardingObserverTest {
 
         store.dispatch(ContentAction.UpdateWebAppManifestAction("1", mockk())).joinBlocking()
         verify(exactly = 0) { settings.incrementVisitedInstallableCount() }
-        verify(exactly = 0) { navigationController.nav(
-            R.id.browserFragment,
-            BrowserFragmentDirections.actionBrowserFragmentToPwaOnboardingDialogFragment())
-        }
+        verify(exactly = 0) { pwaOnboardingObserver.navigateToPwaOnboarding() }
     }
 
     internal class MockedLifecycleOwner(initialState: Lifecycle.State) : LifecycleOwner {

